torch/csrc/distributed/rpc/tensorpipe_agent.h

```diff
@@ -11,12 +11,21 @@
 #include <c10d/Store.hpp>
 #include <torch/csrc/distributed/rpc/rpc_agent.h>
 
+#include <tensorpipe/tensorpipe.h>
+
 // Forward-declare the TensorPipe classes we need, to avoid including its
 // headers in PyTorch's ones and thus have it become a public dependency.
 
 namespace tensorpipe {
 
+#define TENSORPIPE_USE_CUDA_IPC_CHANNEL TENSORPIPE_HAS_CUDA_IPC_CHANNEL && defined(USE_CUDA)
```

Comment:
I believe TENSORPIPE_HAS_CUDA_IPC_CHANNEL will always be false when TENSORPIPE_USE_CUDA is false, which in turn should have the exact same value as PyTorch's USE_CUDA. Which means that this macro could be dropped and replaced by just TP_HAS_CUDA_IPC_CHANNEL.

@beauby Can you confirm?

Reply:
Can `TENSORPIPE_HAS_CUDA_IPC_CHANNEL` be false when `USE_CUDA` is true?

Reply:
No, sorry, I meant it the other way around. Mathematically, `TENSORPIPE_HAS_CUDA_IPC_CHANNEL` implies `USE_CUDA` (because the CUDA IPC channel requires CUDA and couldn't exist without it). The other direction however doesn't hold: you could have `USE_CUDA` _without_ having `TENSORPIPE_HAS_CUDA_IPC_CHANNEL` (for example on Windows, I believe, because IPC handles are Linux-only).